File: backend/Cpp/Calculator.cpp
#include <iostream>
#include <string>
#include <cstdlib>   // for atof

using namespace std;

int main(int argc, char* argv[]) {

    if (argc < 4) {
        cout << "Invalid input";
        return 0;
    }

    double a = atof(argv[1]);
    double b = atof(argv[2]);
    string op = argv[3];

    if (op == "+") cout << a + b;
    else if (op == "-") cout << a - b;
    else if (op == "*") cout << a * b;
    else if (op == "/") {
        if (b == 0) cout << "Divide by zero";
        else cout << a / b;
    }
    else {
        cout << "Invalid operator";
    }

    return 0;
}

File: backend/tools/calculator.js
// backend/tools/calculator.js
module.exports = function (n1, n2, op) {
  n1 = Number(n1);
  n2 = Number(n2);

  if (isNaN(n1) || isNaN(n2)) return "Invalid input";

  switch (op) {
    case "+":
      return n1 + n2;
    case "-":
      return n1 - n2;
    case "*":
      return n1 * n2;
    case "/":
      return n2 === 0 ? "Divide by zero" : n1 / n2;
    default:
      return "Invalid operator";
  }
};
